### PR TITLE
feat(my-pages): Health messaging opening windows

### PR DESCRIPTION
**Pull request by unakb**
## What

Show health center messaging opening hours on the new conversation screen. The intro now includes the recipient's window and expected reply time, an alert explains when sending is blocked (outside opening hours, or the provider doesn't offer messaging) and warns when less than 30 minutes remain until closing. The form is disabled while sending is blocked. 

## Why

Users previously got no explanation when they couldn't message their health center. Now they see why, and when they can try again.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review
- [ ] No AI tools were used in preparing this pull request
- [x] AI tools were used in preparing this pull request
      Tools used: Claude code

## Summary by CodeRabbit

- **New Features**
  - Enhanced the new health conversation flow with dynamic messaging-availability intro (open/close times and expected response timing).
  - Added in-UI messaging availability alerts (closing soon, closed, and messaging not allowed) based on recipient status.
  - Conversation creation now accounts for messaging eligibility by locking/disabling the form when sending isn’t allowed.
- **Bug Fixes**
  - Correctly handles and displays the “awaiting staff reply” blocked reason.
  - Added organization/department information to conversation details and support for team assignees.
- **Tests**
  - Added Jest coverage for messaging window timing and edge cases.

### CODE REVIEW

File: libs/portals/my-pages/health/src/screens/HealthConversations/components/ConversationAvailabilityAlert.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      : blockedReason
+      ? {
+          type: 'warning',
+          title: formatMessage(
+            messages.healthConversationMessagingNotAllowedTitle,
+          ),
+          message: formatMessage(
+            messages.healthConversationMessagingNotAllowedText,
+          ),
+        }
```
[the remark this answers is not in the document]
Comment (unakb) [reply libs/portals/my-pages/health/src/screens/HealthConversations/components/ConversationAvailabilityAlert.tsx:50]: This alert is only for new messages, you're referring to the **reply** blocked enum but we are working with the conversation blocked states here.

File: libs/portals/my-pages/health/src/screens/HealthConversations/components/ConversationAvailabilityAlert.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      : windowInfo.isClosingSoon
+      ? {
+          type: 'warning',
+          title: formatMessage(messages.healthConversationClosingSoonTitle),
+          message: formatMessage(messages.healthConversationClosingSoonText, {
+            openTime: windowInfo.windowOpenLabel,
+            closeTime: windowInfo.windowCloseLabel,
+          }),
```
[the remark this answers is not in the document]
Comment (unakb) [reply libs/portals/my-pages/health/src/screens/HealthConversations/components/ConversationAvailabilityAlert.tsx:59]: Fixed with message that can handle a missing open time

File: libs/portals/my-pages/health/src/screens/HealthConversations/utils/messagingWindow.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+const pad = (value: number) => value.toString().padStart(2, '0')
+
+const formatMinutes = (minutes: number) =>
+  `${pad(Math.floor(minutes / 60))}:${pad(minutes % 60)}`
+
+const parseTimeToMinutes = (time: string): number | undefined => {
+  const match = time.match(/^(\d{1,2}):(\d{2})/)
+  if (!match) return undefined
+  return Number(match[1]) * 60 + Number(match[2])
+}
```
Comment (thorkellmani) [inline libs/portals/my-pages/health/src/screens/HealthConversations/utils/messagingWindow.ts:13]: if this is not already a shared utility function, it should be.

File: libs/portals/my-pages/health/src/screens/HealthConversations/NewHealthConversation.tsx [hunk 4]
```diff
@@ -60,8 +57,32 @@ const NewHealthConversation = () => {
...
+
+  const isFormLocked = recipient?.canCreateConversation === false
+
   const canSubmit =
-    !!selectedTypeCode && !!messageText.trim() && termsAccepted && !sending
+    !!selectedTypeCode &&
+    !!messageText.trim() &&
+    termsAccepted &&
+    !sending &&
+    !isFormLocked
```
Comment (thorkellmani) [inline libs/portals/my-pages/health/src/screens/HealthConversations/NewHealthConversation.tsx:85]: oof, hellish to debug

**Review by thorkellmani — APPROVED** [review]
LGTM ( mostly)

File: libs/portals/my-pages/health/src/screens/HealthConversations/utils/messagingWindow.ts (hunk 3 above)
Comment (unakb) [reply libs/portals/my-pages/health/src/screens/HealthConversations/utils/messagingWindow.ts:13]: I'll move them to the dateUtils

File: libs/portals/my-pages/health/src/screens/HealthConversations/NewHealthConversation.tsx (hunk 4 above)
Comment (unakb) [reply libs/portals/my-pages/health/src/screens/HealthConversations/NewHealthConversation.tsx:85]: It is a lot of conditions, they're all needed and valid though. I moved the first 3 in to a separate collective check called isFormValid, should be a bit more readable 😃
